api/pkg/apis/v1alpha1/utils/symphony-api.go

```diff
@@ -67,6 +68,70 @@ func GetSymphonyAPIAddressBase() string {
 	return symphonyAPIAddressBase
 }
 
+var getApiClientOnce sync.Once
+var apiClient *APIClient
+var err error
+
+func GetApiClient() (*APIClient, error) {
+	getApiClientOnce.Do(func() {
+		apiClient, err = getApiClient()
+	})
+	return apiClient, err
+}
+
+// For testing purpose only
+func UpdateApiClientUrl(url string) {
+	if apiClient == nil {
+		GetApiClient()
+	}
+	apiClient.baseUrl = url
```

Comment:
I can understand baseUrl needs to be changed for testing purpose. But will this cause data concurrency issue? Given baseUrl is not protected.